src/controllers/url.ts

```diff
@@ -0,0 +1,86 @@
+import { Request, Response } from 'express';
+import { customAlphabet } from 'nanoid';
+import Url from '../models/url';
+import { BASE_URL, HTTP_ERRORS } from '../constant';
+import logger from '../logger/index';
+
+const createShortUrls = async (req: Request, res: Response) => {
+  try {
+    const { url, customName } = req.body;
+    let shortUrls;
+    const generateShortId = customAlphabet(
+      'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890',
+      5
+    );
+
+    if (!url) {
+      return res.status(HTTP_ERRORS.BAD_REQUEST).json({ error: 'Original Url is required' });
+    }
+    const existingLongUrl = await Url.findOne({ originalUrl: url });
+    if (existingLongUrl) {
+      return res
+        .status(200)
+        .json({ message: 'Shortened url already exists', shorturl: existingLongUrl.shortUrl });
+    }
+
+    if (customName) {
+      const trimmedName = customName.split(' ').join('');
+      if (trimmedName.length < 5) {
+        return res
+          .status(HTTP_ERRORS.BAD_REQUEST)
+          .json({ error: 'Custom name must be at least five letters' });
+      }
+      const customNamewithDashes = customName.split(' ').join('-');
+      shortUrls = `${BASE_URL}/${customNamewithDashes}`;
+      const existingCustomName = await Url.findOne({ customName });
+      if (existingCustomName) {
+        return res.status(HTTP_ERRORS.BAD_REQUEST).json({ error: 'Custom name already exists' });
+      }
+    } else {
+      const shortId = generateShortId();
+      shortUrls = `${BASE_URL}/${shortId}`;
+    }
+
+    await Url.create({
+      originalUrl: url,
+      shortUrl: shortUrls,
+      customName
+    });
+    return res
+      .status(HTTP_ERRORS.CREATED)
+      .json({ message: 'Url shortened successfully', data: { shorturl: shortUrls } });
+  } catch (error) {
+    logger.error(`Error creating short url: ${error}`);
+    return res.status(HTTP_ERRORS.INTERNAL_SERVER_ERROR).json({ error: 'Internal Server Error' });
+  }
+};
+
+const getAllUrls = async (req: Request, res: Response) => {
+  try {
+    const urls = await Url.find({}, { __v: 0 }).lean();
```

Comment:
If you want to remove the `_v`, it is better you remove it within the model so you don't have to repeat `{ __v: 0 })` everytime you are execute `find`.

I also, want you to remove the underscore from the id. So, instead of `_id`, we have `id`

Reply:
Do you mean in the response objects? or I should remove the _ from id in the database?

Reply:
Only from the response object. Do something like this.

<img width="702" alt="Screenshot 2024-05-04 at 11 40 10" src="https://github.com/PraiseTy/shortit-ts/assets/48496855/8f6dde4e-aec4-4d8d-a859-31ce7e580f79">